src/storage/mvcc/reader.rs

```diff
@@ -82,23 +84,32 @@ impl MvccReader {
         }
 
         let k = key.append_ts(ts);
-        let res = if let Some(ref mut cursor) = self.data_cursor {
-            match cursor.get(&k, &mut self.statistics.data)? {
-                None => panic!("key {} not found, ts {}", key, ts),
-                Some(v) => v.to_vec(),
+        let res = if use_seek && self.data_cursor.is_some() {
```

Comment:
In my opinion, `data_cursor` only create when scan mode is sth, can you give more explanation for this optimization?